src/foam/nanos/auth/CapabilityAuthService.js

```diff
@@ -55,12 +119,20 @@ foam.CLASS({
         if ( x.get(Session.class) == null ) return false;
         if ( user == null || ! user.getEnabled() ) return false;
 
-        // Check whether user has permission to check user permissions.
-        if ( ! getDelegate().check(x, "service.auth.checkUser") ) return false;
+        this.initialize(x);
+
+        String key = user.getId() + permission;
+        Boolean result = ( (Map<String, Boolean>) getCache() ).get(key);
+        if ( result != null ) return result;
```

Comment:
should this be checking the delegate too?

Reply:
Yes added now, thanks